docs/specs/servers/push/spec.md

```diff
@@ -80,27 +94,53 @@ DELETE <PUSH_SERVER_URL>/clients/<CLIENT_ID>
 
 ### Send Notification
 
+This endpoint is called by the relay.
+
 ```
 POST <PUSH_SERVER_URL>/clients/<CLIENT_ID>
+```
+
+- `CLIENT_ID` - This is used to map to the device token
+
+Headers:
+- `X-Ed25519-Timestamp` - timestamp of the signature
+- `X-Ed25519-Signature` - hex encoded ed25519 signature of `format!("{}.{}.{}", timestamp, body.len(), body)` where `body` is the request body
+
+Request body:
+```typescript
 {
-    "id": "0000-0000-0000-0000"
-    "payload": {
-      "topic": "<topic>",
-      "flags": <flags>,
-      "blob": "<blob>"
-    }
-```
-
-- `CLIENT_ID`: This is used to map to the device token
-- `id`: This is generated by the Relay to prevent duplicate notifications
-- `payload`: A string with 3 values encoded and split with a `:` as a delimiter.
-  - `topic`: This is used by the SDKs to decrypt encrypted payloads on the client side
-  - `flags`: an integer value that uses bitmasks to encode information about the payload for the SDKs, the flags are below
-    ```
-    Encrypted  = 1 << 0
-    Sign       = 1 << 1
-    Auth       = 1 << 2
-    Chat       = 1 << 3
-    Notify     = 1 << 4
-    ```
-  - `blob`: If the payload is encrypted this is just the encrypted blob, when this is cleartext it should be base64 encoded. The blob should follow the format from of the [Notify Message](https://specs.walletconnect.com/2.0/specs/clients/notify/data-structures#notify-message)
+  // Topic of the message. This is used by the SDKs to decrypt encrypted payloads on the client side
+  topic: string,
+
+  // Tag of the message.
+  tag: string,
+
+  // The relay message.
+  message: string,
+
+  // ==== Deprecated fields ====
+
+  // Unique ID for deduplication purposes
+  id: string,
+
+  payload: {
+    // Same as `topic` above
+    topic: string,
+
+    // See below
+    flags: number,
+
+    // See below
+    blob: string,
+  }
```

Comment:
@chris13524 according to FCM specs, all the fields and objects must be included inside the data object:
https://firebase.google.com/docs/cloud-messaging/concept-options#data_messages

Otherwise, we cannot access them on the SDK side once received

Reply:
This is for the webhook from relay to push server. The notification content is specified under the Register Client section: https://github.com/WalletConnect/walletconnect-specs/pull/147/files/1a9ad6005447d6d7d6458824579ecc103f4af825#diff-25490e5dda4bcd46802fca4131439d45ab85cf408683b76c7604735dfe656edcR80